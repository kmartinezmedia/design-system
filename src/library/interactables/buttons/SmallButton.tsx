import React, { FunctionComponent } from 'react';
import {
  TouchableWithoutFeedback,
  GestureResponderEvent,
  Animated,
  StyleSheet,
  Text,
} from 'react-native';
import { View, ViewProps } from '@designSystem/primitives';
import { usePressAnimation, useSmallButtonVariant } from '@designSystem/hooks';
import {
  border,
  spacing,
  fontFamily,
  fontSize,
  fontWeight,
} from '@designSystem/theme';
import { SmallButtonVariant } from '@designSystem/types';

export interface SmallButtonProps extends Omit<ViewProps, 'onPress'> {
  disabled?: boolean;
  variant?: SmallButtonVariant;
  onPress?: (event: GestureResponderEvent) => void;
  testID?: string;
}

export const SmallButton: FunctionComponent<SmallButtonProps> = ({
  children,
  disabled,
  onPress,
  testID,
  variant = 'primary',
  ...props
}) => {
  const [pressIn, pressOut, scale] = usePressAnimation();
  const variantStyles = useSmallButtonVariant(variant);
  return (
    <View {...props} alignSelf="flex-start">
      <TouchableWithoutFeedback
        onPress={onPress}
        onPressIn={pressIn}
        onPressOut={pressOut}
        disabled={disabled}>
        <Animated.View
          style={[
            styles.smallButton,
            disabled && styles.disabled,
            { transform: [{ scale }] },
            variantStyles,
          ]}
          testID={testID}>
          <Text style={[styles.text, { color: variantStyles.textColor }]}>
            {children}
          </Text>
        </Animated.View>
      </TouchableWithoutFeedback>
    </View>
  );
};

const styles = StyleSheet.create({
  smallButton: {
    alignItems: 'center',
    justifyContent: 'center',
    borderStyle: 'solid',
    borderRadius: border.buttonRadius,
    paddingVertical: spacing[2],
    paddingHorizontal: spacing[4],
  },
  disabled: {
    opacity: 0.5,
  },
  text: {
    fontFamily: fontFamily.medium,
    fontSize: fontSize.body,
    fontWeight: fontWeight.medium,
  },
});
