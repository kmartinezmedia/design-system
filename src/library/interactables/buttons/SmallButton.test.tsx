import React from 'react';
import 'react-native';
import { create, act, ReactTestRenderer } from 'react-test-renderer';
import { SmallButton } from './SmallButton';

describe('SmallButton', () => {
  it('default', () => {
    let component: ReactTestRenderer;
    act(() => {
      component = create(
        <>
          <SmallButton>Hey, lil 🐛</SmallButton>
          <SmallButton variant="secondary">Hey, lil 🐛</SmallButton>
          <SmallButton variant="positive">Hey, lil 🐛</SmallButton>
        </>,
      );
    });
    expect(component!.toJSON()).toMatchSnapshot();
  });
  it('positions', () => {
    let component: ReactTestRenderer;
    act(() => {
      component = create(
        <>
          <SmallButton spacing={3}>Hey, lil 🐛</SmallButton>
          <SmallButton spacing={2} pin="right">
            Hey, lil 🐛
          </SmallButton>
        </>,
      );
    });
    expect(component!.toJSON()).toMatchSnapshot();
  });
});
