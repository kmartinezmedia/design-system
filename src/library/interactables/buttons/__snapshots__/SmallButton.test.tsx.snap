// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SmallButton default 1`] = `
Array [
  <View
    style={
      Array [
        Object {
          "alignItems": "stretch",
          "alignSelf": "flex-start",
          "flexDirection": "column",
          "flexGrow": undefined,
          "flexShrink": undefined,
          "flexWrap": "nowrap",
          "height": undefined,
          "justifyContent": "flex-start",
          "width": undefined,
        },
        undefined,
        undefined,
        undefined,
        Object {},
      ]
    }
  >
    <TouchableWithoutFeedback
      onPressIn={[Function]}
      onPressOut={[Function]}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderColor": "#F1F3F5",
            "borderRadius": 8,
            "borderStyle": "solid",
            "borderWidth": 1,
            "justifyContent": "center",
            "paddingHorizontal": 16,
            "paddingVertical": 8,
            "textColor": "#1652F0",
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Graphik-Medium",
                "fontSize": 16,
                "fontWeight": "500",
              },
              Object {
                "color": "#1652F0",
              },
            ]
          }
        >
          Hey, lil 🐛
        </Text>
      </View>
    </TouchableWithoutFeedback>
  </View>,
  <View
    style={
      Array [
        Object {
          "alignItems": "stretch",
          "alignSelf": "flex-start",
          "flexDirection": "column",
          "flexGrow": undefined,
          "flexShrink": undefined,
          "flexWrap": "nowrap",
          "height": undefined,
          "justifyContent": "flex-start",
          "width": undefined,
        },
        undefined,
        undefined,
        undefined,
        Object {},
      ]
    }
  >
    <TouchableWithoutFeedback
      onPressIn={[Function]}
      onPressOut={[Function]}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderColor": "#F1F3F5",
            "borderRadius": 8,
            "borderStyle": "solid",
            "borderWidth": 1,
            "justifyContent": "center",
            "paddingHorizontal": 16,
            "paddingVertical": 8,
            "textColor": "#1652F0",
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Graphik-Medium",
                "fontSize": 16,
                "fontWeight": "500",
              },
              Object {
                "color": "#1652F0",
              },
            ]
          }
        >
          Hey, lil 🐛
        </Text>
      </View>
    </TouchableWithoutFeedback>
  </View>,
  <View
    style={
      Array [
        Object {
          "alignItems": "stretch",
          "alignSelf": "flex-start",
          "flexDirection": "column",
          "flexGrow": undefined,
          "flexShrink": undefined,
          "flexWrap": "nowrap",
          "height": undefined,
          "justifyContent": "flex-start",
          "width": undefined,
        },
        undefined,
        undefined,
        undefined,
        Object {},
      ]
    }
  >
    <TouchableWithoutFeedback
      onPressIn={[Function]}
      onPressOut={[Function]}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#00A45F",
            "borderColor": "#00A45F",
            "borderRadius": 8,
            "borderStyle": "solid",
            "borderWidth": 1,
            "justifyContent": "center",
            "paddingHorizontal": 16,
            "paddingVertical": 8,
            "textColor": "#FFFFFF",
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Graphik-Medium",
                "fontSize": 16,
                "fontWeight": "500",
              },
              Object {
                "color": "#FFFFFF",
              },
            ]
          }
        >
          Hey, lil 🐛
        </Text>
      </View>
    </TouchableWithoutFeedback>
  </View>,
]
`;

exports[`SmallButton positions 1`] = `
Array [
  <View
    style={
      Array [
        Object {
          "alignItems": "stretch",
          "alignSelf": "flex-start",
          "flexDirection": "column",
          "flexGrow": undefined,
          "flexShrink": undefined,
          "flexWrap": "nowrap",
          "height": undefined,
          "justifyContent": "flex-start",
          "width": undefined,
        },
        Object {
          "padding": 12,
        },
        undefined,
        undefined,
        Object {},
      ]
    }
  >
    <TouchableWithoutFeedback
      onPressIn={[Function]}
      onPressOut={[Function]}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderColor": "#F1F3F5",
            "borderRadius": 8,
            "borderStyle": "solid",
            "borderWidth": 1,
            "justifyContent": "center",
            "paddingHorizontal": 16,
            "paddingVertical": 8,
            "textColor": "#1652F0",
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Graphik-Medium",
                "fontSize": 16,
                "fontWeight": "500",
              },
              Object {
                "color": "#1652F0",
              },
            ]
          }
        >
          Hey, lil 🐛
        </Text>
      </View>
    </TouchableWithoutFeedback>
  </View>,
  <View
    style={
      Array [
        Object {
          "alignItems": "stretch",
          "alignSelf": "flex-start",
          "flexDirection": "column",
          "flexGrow": undefined,
          "flexShrink": undefined,
          "flexWrap": "nowrap",
          "height": undefined,
          "justifyContent": "flex-start",
          "width": undefined,
        },
        Object {
          "padding": 8,
        },
        Object {
          "bottom": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        undefined,
        Object {},
      ]
    }
  >
    <TouchableWithoutFeedback
      onPressIn={[Function]}
      onPressOut={[Function]}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderColor": "#F1F3F5",
            "borderRadius": 8,
            "borderStyle": "solid",
            "borderWidth": 1,
            "justifyContent": "center",
            "paddingHorizontal": 16,
            "paddingVertical": 8,
            "textColor": "#1652F0",
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Graphik-Medium",
                "fontSize": 16,
                "fontWeight": "500",
              },
              Object {
                "color": "#1652F0",
              },
            ]
          }
        >
          Hey, lil 🐛
        </Text>
      </View>
    </TouchableWithoutFeedback>
  </View>,
]
`;
